pkg/k8s/watchers/subscriber/service.go

```diff
@@ -29,30 +29,42 @@ func (l *ServiceList) Register(s ServiceHandler) {
 }
 
 // NotifyAdd notifies all the subscribers of an add event to a service.
-func (l *ServiceList) NotifyAdd(svc *slim_corev1.Service) {
+func (l *ServiceList) NotifyAdd(svc *slim_corev1.Service) []error {
```

Comment:
I think we'll want to have this so that https://github.com/cilium/cilium/issues/15471 can be more robust and we can respond to errors if we need to. In this case, there may not be an immediate use case to using the returned errors, but it gives us flexibility in the future.

Reply:
@christarazi I do think this is the sentiment, to future proof this usage. Say if "ServiceList" was used else where the calling code could react to specific service event handler errors. 

@aanm what are your thoughts on this?

Reply:
Discussed offline